code/datums/diseases/wizarditis.dm

```diff
@@ -87,32 +87,33 @@ STI KALY - blind
 
 
 /datum/disease/wizarditis/proc/teleport()
-	var/list/theareas = get_areas_in_range(80, affected_mob)
-	for(var/area/space/S in theareas)
-		theareas -= S
+	var/list/possible_areas = get_areas_in_range(80, affected_mob)
+	for(var/area/space/S in possible_areas)
+		possible_areas -= S
 
-	if(!theareas||!theareas.len)
+	if(!length(possible_areas))
 		return
 
-	var/area/thearea = pick(theareas)
+	var/area/chosen_area = pick(possible_areas)
 
-	var/list/L = list()
-	for(var/turf/T in get_area_turfs(thearea.type))
-		if(T.z != affected_mob.z) continue
-		if(T.name == "space") continue
+	var/list/teleport_turfs = list()
+	for(var/turf/T in get_area_turfs(chosen_area.type))
+		if(!is_teleport_allowed(T.z) || T.z != affected_mob.z)
```

Comment:
I'm still wondering how the `z` level can be different from the mob's `z` level. And if it should break if it finds a turf which has that. Or if it should just continue and ignore it like in the old code.
I'm leaning towards the continue and ignore part so that the spell at least functions properly and takes the other turfs into consideration

Reply:
you make a good point, space turfs are skipped and there theoretically should not be areas over multiple z-levels (this is bad mapping anyway so it may help us find mapping errors and fix them :eyes:)